# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../ntp/ntp-4.2.6p2-nano.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Patch was taken from fedora but rediffed against 4.2.6p2

original patch: http://cvs.fedoraproject.org/viewvc/devel/ntp/ntp-4.2.4p7-nano.patch?revision=1.2&content-type=text%2Fplain&view=co

Further information and alternative patches:
https://support.ntp.org/bugs/show_bug.cgi?id=888
https://bugs.launchpad.net/ubuntu/+source/ntp/+bug/412242

diff -ruN ntp-4.2.6p2-orig/include/ntp_syscall.h ntp-4.2.6p2/include/ntp_syscall.h
--- ntp-4.2.6p2-orig/include/ntp_syscall.h	2009-12-09 08:36:37.000000000 +0100
+++ ntp-4.2.6p2/include/ntp_syscall.h	2010-07-26 10:07:22.068755685 +0200
@@ -14,6 +14,14 @@
 # include <sys/timex.h>
 #endif
 
+#if defined(ADJ_NANO) && !defined(MOD_NANO)
+#define MOD_NANO ADJ_NANO
+#endif
+
+#if defined(ADJ_TAI) && !defined(MOD_TAI)
+#define MOD_TAI ADJ_TAI
+#endif 
+
 #ifndef NTP_SYSCALLS_LIBC
 #ifdef NTP_SYSCALLS_STD
 # define ntp_adjtime(t)		syscall(SYS_ntp_adjtime, (t))
